src/app/plugins/dog.plugin.ts

```diff
@@ -1,66 +1,133 @@
+          // Filtering out the the breeds that lack any subbreeds
+          if (breedData[breed].length > 0) {
+            const subBreeds = breedData[breed].map((subBreed: string) => `${subBreed} ${breed}`); // flipped the breed and subBreed
+            this._allBreeds = this._allBreeds.concat(subBreeds);
+
+            this._subBreeds.push({
+              breed: breed,
+              subBreed: subBreeds,
+            });
+          }
         });
+
+        // Sorting the subBreed list so the embed looks better
+        this._subBreeds.sort((a: any, b: any) => a.subBreed.length - b.subBreed.length);
       })
       .catch((err) => this.container.loggerService.warn(err));
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    let breed = this._parseBreed(args || []);
+    // Items in the list are reversed, so the input needs reversing.
+    const breed = this._parseInput(args?.reverse() || []);
 
-    if (breed === '' || breed.toLowerCase() === 'random') {
-      breed = this._breeds[this._getRandomInt(this._breeds.length)];
+    if (breed.startsWith('subbreed')) {
+      await message.reply(await this._makeSubBreedEmbed());
+      return;
     }
 
-    if (!breed) {
-      this.container.loggerService.warn(`Dog API down.`);
-      await message.reply(`Sorry, the dog plugin has encountered an error.`);
+    if (breed.startsWith('breed')) {
+      await message.reply(await this._makeBreedEmbed());
       return;
     }
 
-    if (!this._breeds.includes(breed)) {
-      message.reply(`Breeds supported: \n\`\`\`\n${this._breeds.join('\n')}\`\`\``);
-      return;
+    let url = `breed/${breed.replace(' ', '/')}/images/random`; // The / slash is just how the API takes in the sub breeds
+
+    if (breed === '' || breed === 'random') {
+      url = 'breeds/image/random';
     }
 
     await this.container.httpService
-      .get(`${this._API_URL}breeds/${breed}/image`)
+      .get(`${this._API_URL}${url}`)
       .then((response: IHttpResponse) => {
-        if (!!response.data.response.url) {
-          message.reply('', {
-            files: [response.data.response.url],
-          });
+        // notifies the user if their was a problem contacting the server
+        if (Math.floor(response.status / 100) != 2) {
+          message.reply(
+            `Something seems to have happened with the connection to ${this._API_URL}.`
+          );
+          return;
         }
+
+        message.reply('', {
+          files: [response.data.message],
+        });
       })
-      .catch((err) => this.container.loggerService.warn(err));
+      .catch((err) => {
+        this.container.loggerService.warn(err);
+      });
   }
 
-  private _getRandomInt(max: number): number {
-    return Math.floor(Math.random() * Math.floor(max));
+  async _makeBreedEmbed(): Promise<MessageEmbed> {
+    if (this._breedEmbed) {
+      return this._breedEmbed;
+    }
+
+    this._breedEmbed = new MessageEmbed();
```

Comment:
remove this line

Reply:
_breedEmbed is undefined prior to here through, and if I define it before hand then the if becomes `if (this._breedEmbed === new MessageEmbed())` which I think is less readable.

Reply:
Am I missing something?
You are redefining it on the next line, so that should matter

Reply:
oh wait nope I'm just dumb mb